include/librealsense2/dds/dds-device-broadcaster.cpp

```diff
@@ -241,16 +237,16 @@ bool dds_device_broadcaster::create_device_writer( const std::string &device_key
     return _device_handle_by_sn[device_key].data_writer != nullptr;
 }
 
-bool dds_device_broadcaster::create_dds_publisher()
+void dds_device_broadcaster::create_dds_publisher()
 {
     // Registering the topic type enables topic instance creation by factory
-    _topic_type_ptr->register_type( _participant );
-    _publisher = _participant->create_publisher( PUBLISHER_QOS_DEFAULT, nullptr );
-    _topic = _participant->create_topic( librealsense::dds::topics::device_info::TOPIC_NAME,
-                                         (*_topic_type_ptr)->getName(),
-                                         TOPIC_QOS_DEFAULT );
-
-    return ( _topic != nullptr && _publisher != nullptr );
+    eprosima::fastdds::dds::TypeSupport topic_type( new librealsense::dds::topics::device_info::type );
+    DDS_API_CALL( _participant->register_type( topic_type ) );
```

Comment:
I know this works, but didn't we say we'd leave it as `topic_type.register_type( _participant )`?

Reply:
I thought we said align the call format to all other `FastDDS` API's no?